packages/zod/src/index.ts

```diff
@@ -0,0 +1,341 @@
+import {
+  ParameterObject,
+  PathItemObject,
+  ReferenceObject,
+  RequestBodyObject,
+  ResponseObject,
+  SchemaObject,
+} from 'openapi3-ts';
+import {
+  ClientBuilder,
+  ClientGeneratorsBuilder,
+  escape,
+  GeneratorDependency,
+  GeneratorOptions,
+  GeneratorVerbOptions,
+  isString,
+  resolveRef,
+} from '@orval/core';
+
+const ZOD_DEPENDENCIES: GeneratorDependency[] = [
+  {
+    exports: [
+      {
+        name: '* as zod',
+        alias: 'zod',
+        default: true,
+        values: true,
+        syntheticDefaultImport: true,
+      },
+    ],
+    dependency: 'zod',
+  },
+];
+
+export const getZodDependencies = () => ZOD_DEPENDENCIES;
+
+const resolveZodType = (schemaTypeValue: SchemaObject['type']) => {
+  switch (schemaTypeValue) {
+    case 'integer':
+      return 'number';
+    case 'null':
+      return 'mixed';
+    default:
+      return schemaTypeValue ?? 'mixed';
+  }
+};
+
+const generateZodValidationSchemaDefinition = (
+  schema: SchemaObject | undefined,
+  _required: boolean | undefined,
+) => {
+  if (!schema) return [];
+
+  const validationFunctions: [string, any][] = [];
+  const type = resolveZodType(schema?.type);
+  const required =
+    schema?.default !== undefined
+      ? false
+      : _required ?? !schema?.nullable ?? false;
+  const min =
+    schema?.minimum ??
+    schema?.exclusiveMinimum ??
+    schema?.minLength ??
+    undefined;
+  const max =
+    schema?.maximum ??
+    schema?.exclusiveMaximum ??
+    schema?.maxLength ??
+    undefined;
+  const matches = schema?.pattern ?? undefined;
```

Comment:
There is also a format that can be set in the schema that controls different things but I guess it may be hard to transform that to a validation function. Maybe good to add as a simple export string const though.

Reply:
yes, specifically "date-time" can use the new `.datetime()` function in zod 3.2